### PR TITLE
Tiflash perf tuning add examples

### PR DESCRIPTION
**Pull request by yibin87**
### What is changed, added or deleted? (Required)

Add tuning examples for TiFlash.
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tune-tiflash-performance.md [hunk 1]
```diff
@@ -44,6 +93,58 @@ set @@tidb_enforce_mpp = ON;
 set @@tidb_opt_agg_push_down = ON;
 ```
 
+在以下示例中，`tidb_enforce_mpp` 开启前，TiDB 需要从 TiKV 读取数据，并在 TiDB 侧执行 `Join` 和 `Aggregation`。`tidb_enforce_mpp` 开启后，`Join` 和 `Aggregation` 被下推到了 TiFlash。需要注意的是，有时候优化器生成的执行计划并不稳定，可能生成 `MPP`，也可能不生成，这时候可以开启 `tidb_enforce_mpp`，强制优化器生成 `MPP` 计划。
```
Comment (windtalker) [inline tiflash/tune-tiflash-performance.md:96]: 这段解释写错了？应该是针对 `tidb_opt_agg_push_down` 的吧？

Comment (yibin87) [reply tiflash/tune-tiflash-performance.md]: 改好了

**Review by windtalker — APPROVED** [review]
LGTM

File: tiflash/tune-tiflash-performance.md [hunk 2]
```diff
@@ -44,6 +93,58 @@ set @@tidb_enforce_mpp = ON;
...
+|               └─TableFullScan_23         | 600000000.00 | 600000000 | mpp[tiflash] | table:t2      | tiflash_task:{proc max:63.8ms, min:0s, avg: 21.3ms, p80:63.8ms, p95:63.8ms, iters:9388, tasks:3, threads:60}, tiflash_scan:{dtfile:{total_scanned_packs:73834, total_skipped_packs:439, total_scanned_rows:600007458, total_skipped_rows:3553131, total_rs_index_load_time: 12ms, total_read_time: 270ms}, total_create_snapshot_time: 0ms}   | keep order:false                                                                                   | N/A    | N/A  |
++------------------------------------------+--------------+-----------+--------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------+--------+------+
+13 rows in set (2.15 sec)
+
+mysql> set @@tidb_opt_agg_push_down = ON;
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:119]: ditto

File: tiflash/tune-tiflash-performance.md [hunk 3]
```diff
@@ -54,6 +155,36 @@ TiFlash 暂时还不支持部分可接受 `Distinct` 列的聚合函数，比如
 set @@tidb_opt_distinct_agg_push_down = ON;
 ```
 
+在以下示例中，`tidb_opt_distinct_agg_push_down` 开启前，TiDB 需要从 TiFlash 读取所有数据，并在 TiDB 侧执行 `distinct`。`tidb_opt_distinct_agg_push_down` 开启后，`distinct a` 被下推到了 TiFlash，在 `HashAgg_6` 里新增里一个 `group by` 列 `test.t.a`。这边的两个 warnings 是警告聚合函数不能完全下推。
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:158]: "这边的两个 warnings" 指代不太清晰，建议优化一下这里的描述。

File: tiflash/tune-tiflash-performance.md [hunk 4]
```diff
@@ -54,6 +155,36 @@ TiFlash 暂时还不支持部分可接受 `Distinct` 列的聚合函数，比如
...
+|     └─TableFullScan_10     | 600000000.00 | 600000000 | cop[tiflash] | table:t       | tiflash_task:{proc max:9ms, min:531µs, avg: 4.49ms, p80:5.55ms, p95:6.74ms, iters:9390, tasks:1173, threads:1173}, tiflash_scan:{dtfile:{total_scanned_packs:73834, total_skipped_packs:1231, total_scanned_rows:600010914, total_skipped_rows:9988978, total_rs_index_load_time: 0ms, total_read_time: 16ms}, total_create_snapshot_time: 0ms}  | keep order:false                                       | N/A      | N/A     |
++----------------------------+--------------+-----------+--------------+---------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------+----------+---------+
+4 rows in set, 2 warnings (2 min 23.21 sec)
+
+mysql> set @@tidb_opt_distinct_agg_push_down = ON;
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:172]: ditto

File: tiflash/tune-tiflash-performance.md [hunk 5]
```diff
@@ -82,6 +213,47 @@ ALTER TABLE employees COMPACT PARTITION pNorth, pEast TIFLASH REPLICA;
...
++------------------------------------------+--------------+-----------+--------------+----------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------+---------+------+
+11 rows in set (3.83 sec)
+
+mysql> set @@tidb_broadcast_join_threshold_size = 10000000;
+Query OK, 0 rows affected (0.00 sec)
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:238]: ditto

File: tiflash/tune-tiflash-performance.md [hunk 6]
```diff
@@ -92,6 +264,45 @@ ALTER TABLE employees COMPACT PARTITION pNorth, pEast TIFLASH REPLICA;
 set @@tidb_max_tiflash_threads = 20;
 ```
 
+在以下示例中，`tidb_max_tiflash_threads` 设置前，所有 TiFlash 的算子的执行线程数为24。`tidb_max_tiflash_threads` 设置后，所有 TiFlash 的算子的执行线程数为60。
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:267]:
```suggestion
在以下示例中，`tidb_max_tiflash_threads` 设置前，所有 TiFlash 的算子的执行线程数为 24。`tidb_max_tiflash_threads` 设置后，所有 TiFlash 的算子的执行线程数为 60。
```

File: tiflash/tune-tiflash-performance.md [hunk 7]
```diff
@@ -92,6 +264,45 @@ ALTER TABLE employees COMPACT PARTITION pNorth, pEast TIFLASH REPLICA;
...
++--------------------------------------+--------------+-----------+--------------+---------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------+--------+------+
+9 rows in set (0.67 sec)
+
+mysql> set @@tidb_max_tiflash_threads = 20;
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:286]: ditto

File: tiflash/tune-tiflash-performance.md [hunk 8]
```diff
@@ -101,3 +312,38 @@ set @@tidb_max_tiflash_threads = 20;
 ```sql
 set @@tiflash_fine_grained_shuffle_stream_count = 20;
 ```
+
+在以下示例中，`tiflash_fine_grained_shuffle_stream_count` 设置前，`[ExchangeSender_11, ExchangeReceiver_12, Sort_13, Window_22]` 的 `stream_count` 是8。`tiflash_fine_grained_shuffle_stream_count` 设置后，`[ExchangeSender_11, ExchangeReceiver_12, Sort_13, Window_22]` 的 `stream_count` 是20。
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:316]:
```suggestion
在以下示例中，`tiflash_fine_grained_shuffle_stream_count` 设置前，`[ExchangeSender_11, ExchangeReceiver_12, Sort_13, Window_22]` 的 `stream_count` 是 8。`tiflash_fine_grained_shuffle_stream_count` 设置后，`[ExchangeSender_11, ExchangeReceiver_12, Sort_13, Window_22]` 的 `stream_count` 是 20。
```

File: tiflash/tune-tiflash-performance.md [hunk 9]
```diff
@@ -101,3 +312,38 @@ set @@tidb_max_tiflash_threads = 20;
...
++----------------------------------+--------------+-----------+--------------+---------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------+--------+------+
+7 rows in set (4 min 30.59 sec)
+
+mysql> set @@tiflash_fine_grained_shuffle_stream_count = 20;
```
Comment (shichun-0415) [inline tiflash/tune-tiflash-performance.md:333]: ditto

File: tiflash/tune-tiflash-performance.md [hunk 10]
```diff
@@ -34,6 +34,55 @@ MPP 执行计划可以充分利用分布式计算资源，从而显著提高批
...
++--------------------------------------+--------------+-----------+-----------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+------+
+10 rows in set (22.82 sec)
+
+mysql> set @@tidb_enforce_mpp = ON;
```
[the remark this answers is not in the document]
Comment (yibin87) [reply tiflash/tune-tiflash-performance.md:57]: copy 过去没用的，我这边没有建表语句和数据，只是提供一个参考例子。我将前后的部分分开好了，语句和结果就不分开了。

File: tiflash/tune-tiflash-performance.md (hunk 3 above)
Comment (yibin87) [reply tiflash/tune-tiflash-performance.md]: 好的

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Pull request merged**